feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-page-layout-handler.ts

```diff
@@ -12,72 +12,122 @@ import {
     pageTemplate?: string,
     section?: string
   ) {
     if (
       pageTemplate === CpqConfiguratorPageLayoutHandler.templateName &&
-      section === 'header'
+      section === CpqConfiguratorPageLayoutHandler.sectionHeader
     ) {
-      this.configuratorRouterExtractorService
-        .extractRouterData()
-        .pipe(
-          switchMap((routerData) =>
-            this.breakpointService.isUp(BREAKPOINT.lg).pipe(
-              map((isLargeResolution) => ({
-                isLargeResolution: isLargeResolution,
-                routerData,
-              }))
-            )
-          ),
-          take(1)
-        )
-        .subscribe((cont) => {
-          slots$ = slots$.pipe(
-            map((slots) => {
-              if (
-                cont.routerData.pageType ===
-                ConfiguratorRouter.PageType.CONFIGURATION
-              ) {
-                const extendedSlots = ['PreHeader'];
-                extendedSlots.push(...slots);
-                return extendedSlots;
-              } else if (cont.routerData.displayOnly) {
-                if (cont.isLargeResolution) {
-                  return this.commonConfiguratorUtilsService.getSlotsFromLayoutConfiguration(
-                    this.layoutConfig,
-                    CpqConfiguratorPageLayoutHandler.templateName,
-                    CpqConfiguratorPageLayoutHandler.sectionDisplayOnlyName,
-                    BREAKPOINT.lg
-                  );
-                } else {
-                  return this.commonConfiguratorUtilsService.getSlotsFromLayoutConfiguration(
-                    this.layoutConfig,
-                    CpqConfiguratorPageLayoutHandler.templateName,
-                    CpqConfiguratorPageLayoutHandler.sectionDisplayOnlyName,
-                    BREAKPOINT.xs
-                  );
-                }
-              } else return slots;
-            })
-          );
-        });
+      this.compileRouterAndResolution().subscribe((cont) => {
+        slots$ = slots$.pipe(
+          map((slots) => {
+            return this.getHeaderSlots(slots, cont);
+          })
+        );
+      });
+    } else if (
+      pageTemplate === CpqConfiguratorPageLayoutHandler.templateName &&
+      section === CpqConfiguratorPageLayoutHandler.sectionNavigation
+    ) {
+      this.compileRouterAndResolution().subscribe((cont) => {
+        slots$ = slots$.pipe(
+          map((slots) => {
+            return this.getNavigationSlots(slots, cont);
+          })
+        );
+      });
     }
-
     return slots$;
   }
+
+  protected compileRouterAndResolution(): Observable<RouterResolution> {
+    return this.configuratorRouterExtractorService.extractRouterData().pipe(
+      switchMap((routerData) =>
+        this.breakpointService.isUp(BREAKPOINT.lg).pipe(
+          map((isLargeResolution) => ({
+            isLargeResolution: isLargeResolution,
+            routerData,
+          }))
+        )
+      ),
+      take(1)
```

Comment:
The Take(1) should not be part of the method.
The caller should specify how often it wants to emit a value.